app/src/main/kotlin/cz/lastaapps/menza/ui/components/AgataWalletButton.kt

```diff
@@ -0,0 +1,267 @@
+    var username by remember { mutableStateOf("") }
+    var password by remember { mutableStateOf("") }
+
+    // Load initial username & password
+    if (!LocalInspectionMode.current) {
+        val saved = AgataWalletCredentials.getSavedCredentials(context)
+        if (saved != null) {
+            username = saved.first
+            password = saved.second
+        }
+    }
+
+
+    Dialog(onDismissRequest = { onDismissRequest(false) }) {
+        Card(
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(32.dp),
+            shape = RoundedCornerShape(16.dp)
+        ) {
+            Column(
+                modifier = Modifier.padding(8.dp)
+            ) {
+                Text(
+                    text = stringResource(R.string.wallet_login_title),
+                    modifier = Modifier
+                        .fillMaxWidth(),
+                    textAlign = TextAlign.Center,
+                    style = MaterialTheme.typography.headlineSmall
+                )
+                Text(
+                    text = stringResource(R.string.wallet_login_subtitle),
+                    modifier = Modifier
+                        .fillMaxWidth(),
+                    textAlign = TextAlign.Center,
+                    style = MaterialTheme.typography.bodySmall
+                )
+                Spacer(
+                    modifier = Modifier.padding(8.dp)
+                )
+                OutlinedTextField(
+                    value = username,
+                    onValueChange = { username = it },
+                    label = { Text(stringResource(R.string.wallet_login_username)) }
+                )
+                OutlinedTextField(
+                    value = password,
+                    onValueChange = { password = it },
+                    label = { Text(stringResource(R.string.wallet_login_password)) },
+                    visualTransformation = PasswordVisualTransformation()
+                )
+                Row(
+                    modifier = Modifier.fillMaxWidth(),
+                    horizontalArrangement = Arrangement.End,
+                ) {
+                    TextButton(
+                        onClick = { onDismissRequest(false) },
+                    ) {
+                        Text(stringResource(R.string.wallet_login_cancel))
+                    }
+                    TextButton(
+                        onClick = {
+                            AgataWalletCredentials.saveCredentials(context, username, password)
+                            onDismissRequest(true)
+                        },
+                        enabled = username.isNotBlank() && password.isNotBlank()
+                    ) {
+                        Text(stringResource(R.string.wallet_login_save))
+                    }
+                }
+            }
+        }
+    }
+}
+
+@Preview
+@Composable
+fun AgataLoginDialogPreview() {
+    AgataLoginDialog {}
+}
```

Comment:
Nechci to psát ke každému bodu, takže to zkusím shrnout tady na konci.

Je tu strašně moc logiky, která do UI tak jak se teď píše na Androidu nepatří - měl bys udělat ViewModel, který v sobě drží stav a umožňuje ho editovat z UI pomocí Intentů, nebo funkcí (já osobně preferuji). ViewModel by se následně měl delegovat získávání a ukládání dat na UseCase a ty to delegovat na nějaké Repository na pozadí, co si sahá na providery (síť, db, disk, cache, ...). Takže efektivně tvoje `AgataWalletCredentials` a `AgataWallet` mají být zakopány hluboko jako providery.

Reply:
Snažím se přiblížit Clean architecture a MVVM

Reply:
Konkretne k tomuto bodu:

Sorry, ale mam 0 zkusenosti z native Android a Kotlinem, umim jenom Flutter, takze tady nic nebudu menit protoze vubec nevim jak by to melo vypadat a nemam zas tolik volneho casu.

Na zbytek se kouknu pozdeji.

Reply:
Jasný, pohoda, chápu a nemám problém. Když jsi ale zmínil Flutter, jakou architekturu používáte pokud má nějaké konkrétní jsméno a pokud je jedna doporučená?